compiler/rustc_trait_selection/src/traits/relationships.rs

```diff
@@ -0,0 +1,89 @@
+use crate::infer::InferCtxt;
+use crate::traits::query::evaluate_obligation::InferCtxtExt;
+use crate::traits::{ObligationCause, PredicateObligation};
+use rustc_infer::traits::TraitEngine;
+use rustc_middle::ty::{self, ToPredicate, TypeFoldable};
+
+pub(crate) fn update<'tcx, T>(
+    engine: &mut T,
+    infcx: &InferCtxt<'_, 'tcx>,
+    obligation: &PredicateObligation<'tcx>,
+) where
+    T: TraitEngine<'tcx>,
+{
+    if let ty::PredicateKind::Trait(predicate) = obligation.predicate.kind().skip_binder() {
+        if predicate.trait_ref.def_id
+            != infcx.tcx.require_lang_item(rustc_hir::LangItem::Sized, None)
+        {
+            // fixme: copy of mk_trait_obligation_with_new_self_ty
```

Comment:
Ah, dropped this comment now. That function doesn't take the right parameters, so it can't preserve constness for example -- there might be a refactoring of that function in order, not sure. It's also private to the error_reporting module...